sapl-pdp-embedded/src/test/java/io/sapl/util/JarCreator.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2017-2024 Dominic Heutelbeck (dominic@heutelbeck.com)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.sapl.util;
+
+import java.io.BufferedOutputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.jar.JarEntry;
+import java.util.jar.JarOutputStream;
+import java.util.stream.Stream;
+
+/**
+ * Utility class for creating plain JAR archives.
+ */
+public final class JarCreator {
+
+    public static URL createPoliciesInJar(String jarFolderReference) throws IOException {
+        return createJarFromResource("policies_in_jar.jar", "/setups/policies_in_jar", jarFolderReference);
+    }
+
+    public static URL createBrokenPoliciesInJar(String jarFolderReference) throws IOException {
+        return createJarFromResource("broken_policies_in_jar.jar", "/setups/broken_config", jarFolderReference);
+    }
+
+    private static void createJar(String jarFilePath, String[] sourcePaths) throws IOException {
+        try (var fos = new FileOutputStream(jarFilePath);
+                var bos = new BufferedOutputStream(fos);
+                var jos = new JarOutputStream(bos)) {
+
+            for (String sourcePath : sourcePaths) {
+                var source = Path.of(sourcePath);
+                addFileToJar(source, source, jos);
+            }
+        }
+    }
+
+    private static URL createJarFromResource(String jarName, String resourcePath, String jarFolderReference)
+            throws IOException {
+        var path = System.getProperty("java.io.tmpdir") + "/" + jarName;
```

Comment:
Please use the JUnit tools for tmpdir support:
E.g., see: https://www.baeldung.com/junit-5-temporary-directory#tmpdir-extension-1
Double check, that die directory is deleted after the tests conclude.

Reply:
Resolving by forwarding the tempdir from the Test method property injection and setting the `CleanupMode.Always`.